File: LeetCode_Algorithm/796(Rotate String)/796.js
var stringEqual = function(s, g) {
  let i = 0;
  while (i < s.length && i < g.length) {
    if (s[i] != g[i])
      return false;
    i++;
  }
  return i == s.length && i == g.length ? true : false
}

var rotateString = function(s, goal) {
  // sol1 66ms 89%
  let i = 0;
  while (i < s.length) {
    if (stringEqual(s,goal)) {
      return true
    }
    s = s.slice(1, ).concat(s.slice(0, 1))
    i++;
  }
  return false
};

File: LeetCode_Algorithm/796(Rotate String)/796.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        # sol1 57ms 41%
        # 하나씩 다 넘겨가면서 테스트
        i = 0
        while (i < len(s)):
            if self.stringEqual(s, goal):
                return True
            s = s[1:] + s[0]
            i += 1
        return False
    def stringEqual(self, s, g):
        i = 0
        while i < len(s) and i < len(g):
            if s[i] == g[i]:
                i += 1
            else:
                break
        return True if i  == len(s) else False
